skglm/tests/test_datafits.py

```diff
@@ -114,5 +116,41 @@ def test_gamma():
     np.testing.assert_allclose(clf.coef_, gamma_results.params, rtol=1e-6)
 
 
+def test_cox():
+    rng = np.random.RandomState(1265)
+    n_samples, n_features = 10, 30
+
+    # generate data
+    X = rng.randn(n_samples, n_features)
+    tm = rng.choice(n_samples*n_features, size=n_samples, replace=True).astype(float)
+    s = rng.choice(2, size=n_samples).astype(float)
+    y = (tm, s)
+
+    # generate dummy w, Xw
+    w = rng.randn(n_features)
+    Xw = rng.randn(n_samples)
```

Comment:
even if dummy and currently harmless this is error prone if we need to modify the code later : use X @ w